src/main/java/ca/bc/gov/chefs/etl/forms/pcd/chc/pcPatientServices/processor/PcdChcPCPSApiResponseProcessor.java

```diff
@@ -0,0 +1,148 @@
+import ca.bc.gov.chefs.etl.forms.pcd.chc.pcPatientServices.model.PCPSChcSubmissionData;
+import ca.bc.gov.chefs.etl.util.CSVUtil;
+import ca.bc.gov.chefs.etl.util.FileUtil;
+import ca.bc.gov.chefs.etl.util.JsonUtil;
+
+public class PcdChcPCPSApiResponseProcessor extends BaseApiResponseProcessor {
+  @SuppressWarnings("unchecked")
+  @Override
+  public void process(Exchange exchange) throws Exception {
+    String payload = exchange.getIn().getBody(String.class);
+    payload = JsonUtil.roundDigitsNumber(payload);
+    ObjectMapper mapper = new ObjectMapper();
+    List<Root> pcpsChcModels = mapper.readValue(payload, new TypeReference<List<Root>>() {
+    });
+    List<PCPSChcSubmission> parsedChcPCPS = parsePCPSRequest(pcpsChcModels);
+
+    validateRecordCount(pcpsChcModels, parsedChcPCPS);
+
+    List<IModel> iModels = (List<IModel>) (List<?>) parsedChcPCPS;
+    Map<String, List<List<String>>> map = CSVUtil.provider(iModels);
+
+    boolean isHeaderAdded = (boolean) exchange.getProperties().get(Constants.IS_HEADER_ADDED);
+    List<String> filesGenerated = FileUtil.writeToCSVFile(map, PCDConstants.PCD_CHC_PCPS_DIR, isHeaderAdded);
+
+    SuccessResponse successResponse = new SuccessResponse();
+    successResponse.setFiles(filesGenerated);
+    exchange.getIn().setBody(mapper.writeValueAsString(successResponse));
+  }
+
+  private List<PCPSChcSubmission> parsePCPSRequest(List<Root> pcpsModels)
+      throws IllegalArgumentException, IllegalAccessException, NoSuchFieldException, SecurityException {
+    List<PCPSChcSubmission> parsedChcPCPS = new ArrayList<>();
+    for (Root root : pcpsModels) {
+      PCPSChcSubmission pcpsSubmission = new PCPSChcSubmission();
+
+      pcpsSubmission.setSubmissionId(root.getForm().getSubmissionId());
+      pcpsSubmission.setCreatedAt(CSVUtil.formatDate(root.getForm().getCreatedAt()));
+      pcpsSubmission.setLateEntry(root.getLateEntry());
+      pcpsSubmission.setSubmitterFullName(root.getForm().getFullName());
+      pcpsSubmission.setSubmitterUserName(root.getForm().getUsername());
+      pcpsSubmission.setSubmitterEmail(root.getForm().getEmail());
+      pcpsSubmission.setSubmissionStatus(root.getForm().getStatus());
+      pcpsSubmission.setSubmissionVersion("" + root.getForm().getVersion());
+      pcpsSubmission.setSubmissionFormName(root.getForm().getFormName());
+      pcpsSubmission.setChcName(root.getChcName());
+      pcpsSubmission.setPcnCommunityName(root.getPcnCommunity());
+      pcpsSubmission.setHealthAuthority(root.getHealthAuthority());
+      pcpsSubmission.setFiscalYear(root.getFiscalYear());
+      pcpsSubmission.setPeriodReported(root.getPeriodReported());
+      pcpsSubmission.setReasonForExceptPeriodRep(root.getReasonForExceptionInPeriodReported());
+      pcpsSubmission.setNotes(root.getDataSubmission().getNotes());
+      List<PCPSChcSubmissionData> pcpsSubmissionDataList = new ArrayList<>();
+
+      for (Integer i = 0; i < 13; i++) {
```

Comment:
Is there any reason we're not using i=1 and i <= 13? It will avoid the repeated (i +1) in your code.

Reply:
Updated to  i=1 and i <= 13.